VCF stats report for whole VCFs

yz533cb opened the pull request:
<!--
# sanger-tol/variantcalling pull request

Many thanks for contributing to sanger-tol/variantcalling!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/sanger-tol/variantcalling/tree/main/.github/CONTRIBUTING.md)
-->

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/sanger-tol/variantcalling/tree/main/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

yz533cb committed:
Remove vcf_stats_report from rundeepvariant/main.nf

yz533cb committed:
Update rundeepvariant/main.nf

sainsachiko commented:
Good job @yz533cb, these are things you could consider to make it better:

1. Change target branch from sanger-tol/variantcalling:main to sanger-tol/variantcalling:dev
2. Consider remove visual_report in meta.yml
3. It seems deepvariant/vcfstatsreport is not added yet. If not yet, we may want too add this to complete the features before any further testing.

yz533cb committed:
Update meta.yml

yz533cb committed:
Update deepvariant_caller.nf

Add vcfstatreport module after merging the per-chunk VCFs

yz533cb committed:
Update deepvariant_caller.nf

yz533cb committed:
Update deepvariant_caller.nf

delete the emit of vcfstatreport

yz533cb committed:
Update deepvariant_caller.nf

Add vcfstatsreport emit

yz533cb committed:
Update deepvariant_caller.nf

Edit input of VCFSTATREPORT

yz533cb committed:
Update deepvariant_caller.nf

Remove vcf_stats_report emit. Add gvcf_stats_report emit

yz533cb committed:
Update deepvariant_caller.nf

Add vcf and gvcf emits

yz533cb committed:
Update deepvariant_caller - emit
Update modules.config - add vcf_stats_report and gvcf_stats_report
Add module - vcfstatsreport

yz533cb committed:
Update modules.config

publishDir change to variant_calling/report

sainsachiko reviewed (approved):
This looks good to me.
